contracts/protocol/facets/ExchangeHandlerFacet.sol

```diff
@@ -1459,4 +1475,63 @@ contract ExchangeHandlerFacet is DisputeBase, BuyerBase, IBosonExchangeHandler {
             if (_tokenId != 0) revert InvalidTokenId();
         }
     }
+
+    /**
+     * @notice Handles DR fee collection from mutualizer or seller's pool
+     *
+     * @param _exchangeId - exchange id
+     * @param _offer - offer struct
+     * @param _disputeTerms - dispute resolution terms
+     * @param _drFeeAmount - amount of DR fee to collect
+     */
+    function handleDRFeeCollection(
+        uint256 _exchangeId,
+        Offer storage _offer,
+        DisputeResolutionTerms storage _disputeTerms,
+        uint256 _drFeeAmount
+    ) internal {
+        address mutualizer = _disputeTerms.mutualizerAddress;
+        address exchangeToken = _offer.exchangeToken;
+        if (mutualizer == address(0)) {
+            // Self-mutualize: take fee from seller's pool
+            FundsLib.decreaseAvailableFunds(_offer.sellerId, _offer.exchangeToken, _drFeeAmount);
+        } else {
+            // Use mutualizer: request fee
+            uint256 balanceBefore = FundsLib.getBalance(exchangeToken);
+
+            // Request DR fee from mutualizer
+            bool success = IDRFeeMutualizer(mutualizer).requestDRFee(
+                _offer.sellerId,
+                _offer.id,
+                _drFeeAmount,
+                exchangeToken,
+                _exchangeId,
+                _disputeTerms.disputeResolverId
+            );
+
+            uint256 balanceAfter = FundsLib.getBalance(exchangeToken);
+
+            uint256 feeTransferred = balanceAfter - balanceBefore;
+
+            if (!success || feeTransferred != _drFeeAmount) {
+                revert BosonErrors.DRFeeMutualizerCannotProvideCoverage();
+            } else {
+                FundsLib.increaseAvailableFundsAndEmitEvent(
```

Comment:
This is actually not consistent with how we otherwise handle the funds in the protocol.
For the funds that are just put in the escrow, we don't track them explicitly (refer commitToOffer or escalateDispute), so no available funds should be increased.  
Adding funds to a protocol entity makes them withdrawable by the protocol admin, which is not ok. Also, the emitted event `FundsReleased` is misleading. 

So at this point, actually nothing has to be done. At most `FundsEncumbered` event should be emitted, but even that is not strictly necessary, since `DRFeeRequested` contains the relevant information.